### ISSUE TITLE
gitstatus freezing on android 10

### ISSUE THREAD

**Issue by usernein** [romkatv/gitstatus#163]
[p10k.log](https://github.com/romkatv/gitstatus/files/4947283/p10k.log)
Recently i changed my phone and now i'm with Android 10. I tried to access a git directory and it just freezed whole terminal. I thought it could be happening due to the size of the directory but it's only 6MB.
So i closed Termux, opened again then deleted the .git folder before entering it. It entered normally, didn't freeze.
Then i ran 'git init' and it freezed again. That was when i realized it occurs with any git directory, even if it's empty and new. I recorded a [video](https://usernein.gramdevs.com/p10k-gitstatus-freezes.mp4) showing it.

I don't have any clues about what can it be, but i mentioned i'm at Android 10 because it's the newest change i made (and the same folder was working on my old phone running android Marshmallow) 

Screenshot of neofetch result:
![](https://usernein.gramdevs.com/osprey-neofetch.jpg)

**Comment by romkatv** [comment]
Please run the following command:

```zsh
( 
  emulate -L zsh -o err_return
  export TMPPREFIX=${TMPDIR:-/tmp}/zsh
  local tmp
  tmp="$(mktemp -d ${TMPDIR:-/tmp}/p10k.XXXXXXXXXX)"
  {
    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git $tmp/powerlevel10k
    >$tmp/.zshrc <<\END
POWERLEVEL9K_MODE=ascii
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(dir vcs)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=()
source $ZDOTDIR/powerlevel10k/powerlevel10k.zsh-theme
END
    ZDOTDIR=$tmp zsh
  } always {
    cd /
    command rm -rf -- $tmp
  }
)
```

When prompt appears, check whether the problem reproduces from this shell.

**Comment by usernein** [comment]
It does
![](http://usernein.gramdevs.com/test1-gitstatus.jpg)

**Comment by usernein** [comment]
Done

**Comment by romkatv** [comment]
Now that's odd. The log indicates that everything works fine. The only difference I see is that this time around the command itself has created a git repo rather than using your `test` repo. What I want is to see gitstatus daemon log when your prompt breaks. Let's try it this way.

Run this command:

```zsh
( 
  emulate -L zsh -o err_return
  export TMPPREFIX=${TMPDIR:-/tmp}/zsh
  local tmp
  tmp="$(mktemp -d ${TMPDIR:-/tmp}/p10k.XXXXXXXXXX)"
  {
    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git $tmp/powerlevel10k
    >$tmp/.zshrc <<\END
POWERLEVEL9K_MODE=ascii
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(dir vcs)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=()
POWERLEVEL9K_VCS_MAX_SYNC_LATENCY_SECONDS=1
GITSTATUS_LOG_LEVEL=DEBUG
source $ZDOTDIR/powerlevel10k/powerlevel10k.zsh-theme
END
    ZDOTDIR=$tmp zsh
  } always {
    cd /
    command rm -rf -- $tmp
  }
)
```

When prompt appears, type this and store the output somewhere:

```zsh
print -r -- $GITSTATUS_DAEMON_LOG_POWERLEVEL9K
```

Go to your `test` repo that breaks prompt. Once prompt is broken, attach the file indicated above to this issue. You can attach it to a comment instead of editing your original issue description.

**Comment by usernein** [comment]
There is it:
[gitstatus.POWERLEVEL9K.10205.21927.1595249532.2.daemon.log](https://github.com/romkatv/gitstatus/files/4947801/gitstatus.POWERLEVEL9K.10205.21927.1595249532.2.daemon.log)
I ran the command and waited for 3 minutes after it freezed, then i copied the file

**Comment by romkatv** [comment]
Thanks!

Apparently Android kills gitstatusd with `SIGSYS`. This means that Android 10 has put restrictions on some system call(s) that gitstutusd needs. I don't have access to Android 10 and I don't feel like spending time or resources getting such access. If you feel comfortable firing up a debugger and figuring out which system call gets denied, I'll see if I can get change gitstautsd so that it don't use it. If not, I'm going to close the issue with the resolution that Android 10 is not supported.

**Comment by romkatv** [comment]
Oh, you are using an armv7l device? This explains why no one else has complained yet. Most Android devices are 64-bit.

The docs at https://android-developers.googleblog.com/2017/07/seccomp-filter-in-android-o.html also have this to say:

> In total, the filter blocks 17 of 271 syscalls in arm64 and 70 of 364 in arm.

So, 19% of all syscalls are blocked on 32-bit while only 6% are blocked on 64-bit. I guess whatever syscalls gitstatusd makes are allowed on 64-bit, so it works fine.

**Comment by romkatv** [comment]
Closing as I don't intend to do anything about this issue. If you figure out which system call gets blocked, please post the info here or open another issue.
